lib/matplotlib/animation.py

```diff
@@ -250,30 +250,56 @@ class MovieWriter(AbstractMovieWriter):
 
     def __init__(self, fps=5, codec=None, bitrate=None, extra_args=None,
                  metadata=None):
-        '''MovieWriter
+        """
+        MovieWriter
 
         Parameters
         ----------
         fps: int
             Framerate for movie.
+
         codec: string or None, optional
             The codec to use. If ``None`` (the default) the ``animation.codec``
             rcParam is used.
+
         bitrate: int or None, optional
             The bitrate for the saved movie file, which is one way to control
             the output file size and quality. The default value is ``None``,
             which uses the ``animation.bitrate`` rcParam.  A value of -1
             implies that the bitrate should be determined automatically by the
             underlying utility.
-        extra_args: list of strings or None, optional
-            A list of extra string arguments to be passed to the underlying
-            movie utility. The default is ``None``, which passes the additional
-            arguments in the ``animation.extra_args`` rcParam.
+
+        extra_args: list of strings, string or None, optional
+            Either a list of extra string arguments to be passed to the
+            underlying movie utility or a full command string
+            explained below. The default is ``None``,which passes the
+            additional arguments in the ``animation.extra_args`` rcParam.
+
         metadata: Dict[str, str] or None
             A dictionary of keys and values for metadata to include in the
             output file. Some keys that may be of use include:
             title, artist, genre, subject, copyright, srcform, comment.
-        '''
+
+        Notes
+        -----
+        *extra_args* can be a string with the full command line argument used
+        in the ffmpeg, imagemagick and avconv writers. The string is then
+        formatted with the python method ``format``. Possible keywords for the
+        pipe based writers are::
+
+            path, fps, delay=100/fps, x_size, y_size (in pixels),
+            frame_format, codec, bitrate, outfile
+
+        and for the file bases writers::
+
+            path, fps, delay=100/fps, frame_format, temp_name,
+            N (number of frames ), codec, bitrate, outfile
+
+        an example command string is::
+
+            'ffmpeg -r {fps} -i _tmp%07d.png -r 25 -y {outfile}'
```

Comment:
That'll cause all sorts of funny bugs if there are spaces in {outfile}.
Also passing a string (rather than a list) to subprocess.run simply doesn't work on Linux (unless you use shell=True, but then you also run into metacharacter issues.

Reply:
The command is passed as a list.

Reply:
Indeed, I stand corrected.  But using str.split *after* str.format will still run into issues if the filename has spaces.